core/utils/icmp.h

```diff
@@ -5,8 +5,8 @@ namespace utils {
 
 // A basic ICMP header definition.
 struct[[gnu::packed]] IcmpHeader {
-  uint16_t type;      // ICMP packet type.
-  uint16_t code;      // ICMP packet type.
+  uint8_t type;       // ICMP packet type.
```

Comment:
Given that this was off and we didn't catch it before, I wonder if there's an easy way for you to add a test / tests to this PR for the various checksum calculations?